### PR TITLE
Make entitylookup grid-relative

### PR DESCRIPTION
**Pull request by metalgearsloth**
As the scope of https://github.com/space-wizards/RobustToolbox/pull/1848 is now a cruise missile and includes shuttles this is the last blocker to having grids move performantly that I was stalling until I was ready.

Client perf has dropped marginally but we can look at fixing it when entity anchoring is stable.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Will push my latest changes that work with strafing shuttles when I get a chance.

**Comment by metalgearsloth** [comment]
Good for review. Probably doesn't work for rotated shuttles (grid fixtures need more work first before playtesting it is easy) but works fine for strafing ones.

File: Robust.Client/GameObjects/Components/RenderingTreeComponent.cs [hunk 1]
```diff
@@ -15,29 +15,43 @@ public sealed class RenderingTreeComponent : Component
         private static Box2 SpriteAabbFunc(in SpriteComponent value)
         {
             var worldPos = value.Owner.Transform.WorldPosition;
-            return new Box2(worldPos, worldPos);
+            var tree = RenderingTreeSystem.GetRenderTree(value.Owner);
+
+            var pos = worldPos - tree?.Owner.Transform.WorldPosition ?? Vector2.Zero;
+
+            return new Box2(pos, pos);
         }
```
Comment (metalgearsloth) [inline Robust.Client/GameObjects/Components/RenderingTreeComponent.cs:23]: I'm not a huge fan of these and I might just end up making the query approx instead so it doesn't do it, depending how it profiles.

**Review by gradientvera — APPROVED** [review]
looks good to me (<- clueless)

**Pull request merged**